packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridLoadIcon } from '../../icons/index';
+import { GridFilterInputValueProps } from './GridFilterInputValueProps';
+
+export type GridFilterInputDateProps = GridFilterInputValueProps &
+  TextFieldProps & { type?: 'date' | 'datetime-local' };
+
+export const SUBMIT_FILTER_DATE_STROKE_TIME = 500;
+
+function GridFilterInputDate(props: GridFilterInputDateProps) {
+  const { item, applyValue, type, apiRef, focusElementRef, ...other } = props;
+
+  const filterTimeout = React.useRef<any>();
+  const [filterValueState, setFilterValueState] = React.useState(item.value ?? '');
+  const [applying, setIsApplying] = React.useState(false);
+  const id = useId();
+
+  const onFilterChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      const value = event.target.value;
+
+      clearTimeout(filterTimeout.current);
+      setFilterValueState(String(value));
+
+      setIsApplying(true);
+      filterTimeout.current = setTimeout(() => {
+        applyValue({ ...item, value });
+        setIsApplying(false);
+      }, SUBMIT_FILTER_DATE_STROKE_TIME);
+    },
+    [applyValue, item],
+  );
+
+  React.useEffect(() => {
+    return () => {
+      clearTimeout(filterTimeout.current);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    const itemValue = item.value ?? '';
+    setFilterValueState(String(itemValue));
+  }, [item.value]);
+
+  const InputProps = {
+    ...(applying ? { endAdornment: <GridLoadIcon /> } : {}),
+    max: type === 'datetime-local' ? '9999-12-31T23:59' : '9999-12-31',
+    ...other.InputProps,
+  };
+
+  return (
+    <TextField
+      id={id}
+      label={apiRef.current.getLocaleText('filterPanelInputLabel')}
+      placeholder={apiRef.current.getLocaleText('filterPanelInputPlaceholder')}
+      value={filterValueState}
+      onChange={onFilterChange}
+      type={type || 'text'}
+      variant="standard"
+      InputLabelProps={{
+        shrink: true,
+      }}
+      inputRef={focusElementRef}
+      {...other}
```

Comment:
This is not a problem here, but per convention `other` should always be last prop to be spread.